test/common/ops/scoreTask.test.js

```diff
@@ -0,0 +1,185 @@
+/* eslint-disable no-shadow */
+import scoreTask from '../../../common/script/ops/scoreTask';
+import {
+  generateUser,
+  generateDaily,
+  generateHabit,
+  generateTodo,
+  generateReward,
+} from '../../helpers/api-integration/v3';
+import common from '../../../common';
+
+let EPSILON = 0.0001; // negligible distance between datapoints
+
+/* Helper Functions */
+let rewrapUser = (user) => {
+  user._wrapped = false;
+  common.wrap(user);
+  return user;
+};
+
+let beforeAfter = async () => {
+  let before = await generateUser();
+  let after = _.cloneDeep(before);
+  rewrapUser(after);
+
+  return {
+    before,
+    after,
+  };
+};
+
+let expectGainedPoints = (before, after, beforeTask, afterTask) => {
+  expect(after.stats.hp).to.eql(50);
+  expect(after.stats.exp).to.be.greaterThan(before.stats.exp);
+  expect(after.stats.gp).to.be.greaterThan(before.stats.gp);
+  expect(afterTask.value).to.be.greaterThan(beforeTask.value);
+  if (afterTask.type === 'habit') {
+    expect(afterTask.history).to.have.length(1);
+  }
+};
+
+let expectClosePoints = (before, after, beforeTask, task) => {
+  expect(Math.abs(after.stats.exp - before.stats.exp)).to.be.lessThan(EPSILON);
+  expect(Math.abs(after.stats.gp - before.stats.gp)).to.be.lessThan(EPSILON);
+  expect(Math.abs(task.value - beforeTask.value)).to.be.lessThan(EPSILON);
+};
+
+let _expectRoughlyEqualDates = (date1, date2) => {
+  expect(date1.toString()).to.eql(date2.toString());
+};
+
+describe('shared.ops.scoreTask', () => {
+  let ref;
+
+  beforeEach(async () => {
+    ref = await beforeAfter();
+  });
+
+  it('throws error not enough gold', async () => {
+    let reward = await generateReward({ userId: ref.after._id, text: 'some reward', value: 100 });
+    expect(scoreTask.bind(_, { user: ref.after, task: reward })).to.throw('NotAuthorized: Not Enough Gold');
```

Comment:
I get `AssertionError: expected [Function] to throw 'NotAuthorized: Not Enough Gold' but 'NotAuthorized: Not Enough Gold' was thrown`, which is the reason why I wrote it like that - because that way the assertion worked. The _ is a placeholder scope because which scope to use does not matter for this test.